admin-dev/themes/new-theme/js/pages/order/view/order-product-cancel.js

```diff
@@ -0,0 +1,143 @@
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2020 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+import Router from '@components/router';
+import OrderViewPageMap from '@pages/order/OrderViewPageMap';
+import { NumberFormatter } from '@app/cldr';
+
+const {$} = window;
+
+/**
+ * manages all product cancel actions, that includes all refund operations
+ */
+export default class OrderProductCancel {
+  constructor() {
+    this.router = new Router();
+    this.cancelProductForm = $(OrderViewPageMap.cancelProduct.form);
+    this.orderId = this.cancelProductForm.data('orderId');
+    this.orderDelivered = parseInt(this.cancelProductForm.data('isDelivered'), 10) === 1;
+    this.isTaxIncluded = parseInt(this.cancelProductForm.data('isTaxIncluded'), 10) === 1;
+    this.discountsAmount = parseFloat(this.cancelProductForm.data('discountsAmount'));
+    this.currencyFormatter = NumberFormatter.build(this.cancelProductForm.data('priceSpecification'));
+  }
+
+  showPartialRefund() {
+    // Always start by hiding elements then show the others, since some elements are common
+    $(OrderViewPageMap.cancelProduct.toggle.standardRefund).hide();
+    $(OrderViewPageMap.cancelProduct.toggle.partialRefund).show();
+    $(OrderViewPageMap.cancelProduct.table.actions).hide();
+    this.initForm(
+      $(OrderViewPageMap.cancelProduct.buttons.save).data('partialRefundLabel'),
+      this.router.generate('admin_orders_partial_refund', {orderId: this.orderId})
+    );
+  }
+
+  showStandardRefund() {
+    // Always start by hiding elements then show the others, since some elements are common
+    $(OrderViewPageMap.cancelProduct.toggle.partialRefund).hide();
+    $(OrderViewPageMap.cancelProduct.toggle.standardRefund).show();
+    $(OrderViewPageMap.cancelProduct.table.actions).hide();
+    this.initForm(
+      $(OrderViewPageMap.cancelProduct.buttons.save).data('standardRefundLabel'),
+      ''
+    );
+  }
+
+  hideRefund() {
+    $(OrderViewPageMap.cancelProduct.toggle.partialRefund).hide();
+    $(OrderViewPageMap.cancelProduct.toggle.standardRefund).hide();
+    $(OrderViewPageMap.cancelProduct.table.actions).show();
+  }
+
+  initForm(actionName, formAction) {
+    this.listenForInputs();
+    this.updateVoucherRefund();
+
+    this.cancelProductForm.attr('action', formAction);
+    $(OrderViewPageMap.cancelProduct.buttons.save).html(actionName);
+    $(OrderViewPageMap.cancelProduct.table.header).html(actionName);
+    $(OrderViewPageMap.cancelProduct.checkboxes.restock).attr('checked', this.orderDelivered);
+    $(OrderViewPageMap.cancelProduct.checkboxes.creditSlip).attr('checked', true);
+  }
+
+  listenForInputs() {
+    $(OrderViewPageMap.cancelProduct.inputs.quantity).off('change').on('change', (event) => {
```

Comment:
Is it mandatory to `off('change')`?

Reply:
It's better because the listeners are added each time we click on the "Display partial refund button" and I don't want an input to have 5 duplicated listeners Besides this needs to be done because some rows may have been added (or removed) via javascript and ajax actions

Of course we could add the listener on the go each time a row is added, but it seems simpler to do it this way and less prone to error We already had a similar discussion about how it prevents some javascript overriding, which is partly true but none of our frontend code is designed for override right now anyway

Reply:
So instead of adding listener each time you click on the displya button. use the document syntax
```suggestion
    $(document).on('change', OrderViewPageMap.cancelProduct.inputs.quantity, (event) => {
```

Reply:
It will be faster than creating an unlimited amount of js events

Reply:
Mmm I like that I'm gonna try!

Reply:
It works!